src/controller/AutoCommissioner.h

```diff
@@ -94,6 +94,21 @@ class AutoCommissioner : public CommissioningDelegate
                  mDeviceCommissioningInfo.network.thread.endpoint != kInvalidEndpointId));
     };
 
+    // Helper function to Determine whether secondary network interface is supported.
+    // Only true if information is provided for both networks, and the target has endpoint
+    // for wifi and thread.
+    bool IsSecondaryNetworkSupported() const
+    {
+        return ((mParams.GetSupportsConcurrentConnection().ValueOr(false) && mParams.GetWiFiCredentials().HasValue() &&
+                 mDeviceCommissioningInfo.network.wifi.endpoint != kInvalidEndpointId) &&
+                mParams.GetThreadOperationalDataset().HasValue() &&
+                mDeviceCommissioningInfo.network.thread.endpoint != kInvalidEndpointId);
+    }
+
+    void SetTrySecondaryNetwork() { mAttemptedSecondaryNetwork = true; }
```

Comment:
OK, so where does mAttemptedSecondaryNetwork (needs to be named better) get set to false again?

Because it sure looks like the first commissioning will work right, but the second will fail to configure networks properly.

Reply:
Added reset for the flag.